internal/store/interfaces.go

```diff
@@ -47,12 +21,52 @@ type Route interface {
 	ID() string
 }
 
-// Store is used for persisting and querying gateways and routes
+type GatewayRecord struct {
+	ID   core.GatewayID
+	Data []byte
+}
+
+type RouteRecord struct {
+	ID   string
+	Data []byte
+}
```

Comment:
These are introduced so we have a simple way of doing bulk "upsert" operations. Since a route/gateway update can potentially affect a bunch of other routes/gateways, rather than updating them all individually, the idea is to pass off to the backend a list of records that need to be updated with:

1. their identifier (most likely used to construct some sort of key to their storage location), and
2. the marshaled bytes of the data to store